src/dblib/dblib.c

```diff
@@ -5955,9 +5971,9 @@ dbgetuserdata(DBPROCESS * dbproc)
  * \ingroup dblib_core
  * \brief Specify a db-lib version level.
  * 
- * \param version anything, really. 
- * \retval SUCCEED Always.  
- * \remarks No effect on behavior of \c db-lib in \c FreeTDS.  
+ * \param version Any DBVERSION_* constant.
+ * \retval SUCCEED if version was valid, FAIL otherwise.
+ * \remarks Use the corresponding protocol version for subsequent connections.
```

Comment:
I think the purpose of dbsetversion is quite different and should not be changed. Protocol version to connect to a single database can be changed with dbsetlversion/DBSETLVERSION. Setting globally seems a mistake, would prevent multiple connections with different versions.

Reply:
FWIW, this patch goes back to the days of Sybase 12.x (or older?), which lacked these APIs.  That said, I'm on board with dropping it at this point.